### DISCUSSION TITLE
Sobre trabalhar com Windows

### DISCUSSION THREAD

**Discussion by pedrohenri** [frontendbr/forum#113] [Discussão]
O Windows realmente tem limitações, ou isso está só na cabeça do desenvolvedor? Você já teve experiências negativas com o Windows no trabalho? Nesse caso, devemos tentar contornar os obstáculos ou é recomendável que o Front-end adote um sistema Unix?

Existe uma outra discussão aberta sobre dicas de ferramentas para desenvolver no windows. Você pode citar elas aqui também, mas o foco desta discussão é a existência ou não de limitações e se existe realmente um SO superior ou recomendado para se trabalhar.

**Comment by felipefialho** [comment]
@pedrohenri Tomei a liberdade de editar seu titulo para ficar menor, beleza? :)

**Comment by icaromh** [comment]
Eu, particularmente, tive muito mais problemas pra configurar ambiente no Windows, sempre.
- setar variáveis de ambiente;
- fazer workarounds pra instalar docker/php/composer. Basicamente qualquer coisa que envolva servidor e não seja feito pela microsoft é um saco;
- terminal: ssh, apt-get, autocomplete, zsh. Trabalho hoje com diversos terminais abertos

Não tenho trabalhado com layouts e, quando, uso o [Avocode](https://avocode.com/) que abre o PSD com as camadas, lista fontes e faz uma caralhada de coisas legais :dancer: 

Agora, se tu tá produtivo lá e sabe fazer esses processos de olhos fechados. O windows é excelente pra ti. Limitações em si eu só vejo da pessoa, em saber e querer trabalhar com o SO.

**Comment by felipefialho** [comment]
Sempre utilizei Windows e continuo utilizando. Nunca tive problemas (a não ser uma vez para configurar o Ruby).

E agora a coisa vai melhorar de vez com a [implementação da linha de comando do Linux no Windows](http://www.theverge.com/2016/3/30/11331014/microsoft-windows-linux-ubuntu-bash)

**Comment by yurimatheus** [comment]
Há S.O's e S.O's rsrsrsrs

Acredito que com o bash no windows a coisa vai mudar mais de figura.

Quem usa Mac e Linux sente bem mais facilidade em usar as ferramentas
disponíveis para front-end visto que a maioria delas são instaláveis via
linha de comando. No caso do Windows a coisa muda mais de figura, é bem
mais difícil acessar executar algumas tarefas.

Como sou usuário linux, quando entrei pra valer no Universo FrontEnd, eu
notei que a maioria dos FrontEnders usavam ou mac ou linux (ubuntu) para
desenvolverem e já que uso ubuntu isso me ajudou a ter uma curva de
aprendizado bem maior! Quando usava o windows na empresa quebrava bem mais
a cabeça para executar certas atividades.

Acredito que quem determina qual o(s) SO's melhores é a Comunidade. Sen a
comunidade cria frameworks e ferramentas que são melhor executadas em
ambiente unix e linux ... Você já tem a sua resposta.

**Comment by raphaklaus** [comment]
Uso Windows faz uns 20 anos (since 3.11), e Linux uns 7 anos... E sinceramente? Sempre preferi o Windows como sistema pessoal. Mas tem certas ferramentas necessárias para um melhor usufruto, tais como: **cygwin** e Cmder. 

Porém para desenvolvimento, não adianta, Linux é **imbatível**.

Com o anúncio do bash pro Windows 10, as coisas vão melhorar por lá.

**Comment by clarabatt** [comment]
Eu sempre usei Windows a minha vida toda e ultimamente vi a necessidade de experimentar o Linux por três motivos: 
1. **Influência das pessoas**, já que muitos dizem que Linux/Mac é muito bom e que Windows é ruim.
2. **Segurança**, questão que foi levantada quando o Windows 10 foi lançado.
3. **$$$Mac$$$**, hoje não dá pra ter um Mac, quem sabe um dia :satisfied:

Fiz um dual boot e comecei a usar, gostei bastante porque o terminal facilita muito a vida. Mas senti falta de muitas ferramentas que uso, como o pacote **Adobe** e principalmente o **OneDrive** que tem uma interação linda e maravilhosa com o Windows além de ter um ótimo custo/benefício.

Tenho uns 6 meses usando e acho que não conseguiria abandonar o Windows completamente, apesar de achar o Ubuntu ótimo ele não me dá tudo que preciso, até o bash chegar provavelmente continuarei usando os dois.

**Comment by felipefialho** [comment]
Falando em bash, tem previsão de quando vai ser implementado?

**Comment by baltazarparra** [comment]
@LFeh pra galera do windows insiders já está liberado.
A partir da build 14316 já dá pra habilitar.

**Comment by felipefialho** [comment]
Tem cara de vir nas próximas atualizações então :)

**Comment by lazarofl** [comment]
Ainda tem muita coisa pra melhorar, tentei instalar o nvm, mongodb e etc e não tive sucesso, parece que esta numa versão ainda bem limitada mas os caras estão trabalhando forte e trazendo gente de peso pra fazer a coisa virar, principalmente na divisão do Azure deles.

http://mspoweruser.com/oracle-executive-wim-coekaerts-has-joined-microsoft-as-cvp-of-open-source-in-the-enterprise-cloud-group/

**Comment by karlprieb** [comment]
@lazarofl Também tentei instalar o nvm, oh-my-zsh e sem sucesso. Inclusive o maior problema são os symlinks que ainda não estão implementados, impossibilitando de usar muita coisa como vários pacotes no NPM :(

**Comment by kazzkiq** [comment]
**Sobre Windows:**
Xodózinho desde a infância, aprendi a mexer no computador com ele, passei muitas horas jogando, modificando, e me familiarizando com o mundo da informática graças à máquina do Tio Bill. Mas do ponto de vista de desenvolvimento tudo é mais difícil. Instalar Node.js, Ruby, Apache, MySQL ou qualquer outra ferramenta geralmente é muito mais complexo e passível de erros do que os sistemas UNIX com seus lindos terminais.

**Sobre Linux:**
Leia-se Ubuntu, porque foi a única distro que realmente usei pra valer. Eu acho um sistema incrível, que não deixa a desejar em ~~quase~~ nada. Seja para uso doméstico ou para desenvolvimento, é muito prático e útil. Só acho que o Ubuntu peca um pouco por ainda pedir certo conhecimento para algumas tarefas  (como atualizar a placa de vídeo, ou ter que instalar drivers com 1001 comandos de terminal), o que pode espantar usuários leigos.

**Sobre Mac OS:**
O melhor dos dois mundos. Rápido, elegante e ~~quase tão~~ fácil de usar como o Windows, mas também é extremamente versátil na hora de trabalhar com desenvolvimento. Pra quase tudo já existe um app de qualidade disponível, e quase toda ferramenta e/ou problema pode ser instalada/resolvido com um (ou alguns) comandos no terminal. O problema é o **preço**. Os apps bons custam caro, o hardware custa caro, e a manutenção (caso necessária) custa caro. Na nossa realidade pobre de terceiro mundo, com moeda fraca, e economia enferrujada, é uma realidade para poucos (a não ser que a pessoa pare de comer e de viver por uns meses até juntar dinheiro para comprar um Mac lá fora, fica a dica)

**Comment by caioincau** [comment]
Alguma novidade sobre esse Bash com Windows? Ia abrir um tópico para falar sobre isso, como andam as coisas para programar em Windows? Sempre tive dificuldade em me adaptar com o terminal deles, fora o problema com variáveis de ambiente.

**Comment by fredmosc** [comment]
tenho usado bastante, o único porém é que não suporta docker ainda, fora
isso é uma mão na roda

Em seg, 11 de set de 2017 às 12:59, Caio Incau <notifications@github.com>
escreveu:

> Alguma novidade sobre esse Bash com Windows? Ia abrir um tópico para falar
> sobre isso, como andam as coisas para programar em Windows? Sempre tive
> dificuldade em me adaptar com o terminal deles, fora o problema com
> variáveis de ambiente.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/113#issuecomment-328575100>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJB79RAq0goyYYFFhhK2tbsHNo6UdqTeks5shVjNgaJpZM4IBIkz>
> .
>
-- 
Att,

FM
<https://about.me/fredmosc?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=edit_panel&utm_content=thumb>
Frederick Moschkowich
about.me/fredmosc
<https://about.me/fredmosc?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=edit_panel&utm_content=thumb>

**Comment by rubenmarcus** [comment]
Ninguem usou vmware , vagrant e etc pra simular outros ambientes no windows?

**Comment by kivervinicius** [comment]
@rubenmarcus eu usava docker mas chega um determinado momento se torna altamente improdutivo. Então faz um bom tempo que adotei o inverso utilizo linux e em um virtualbox uso o windows para tarefas rotineiras (pacote adobe).

**Comment by rubenmarcus** [comment]
@kivervinicius  e não fica lento? o Pacote Adobe ? Eu uso o windows por hábito mesmo, pelo pacote Adobe e pelos games que meu filho joga kkkkk

**Comment by kivervinicius** [comment]
@rubenmarcus como meu fluxo com windows é baixo poucas vezes eu jogo no computador e muitas vezes a parte de design é baixa então compensa, agora para quem utiliza dia a dia acredito que seja bem incomodo.

**Comment by aferreira44** [comment]
Uso o Windows pq atualmente trabalho com desenvolvimento .NET e tbm preciso
do pacote Adobe de vez em quando. Tinha dual boot com Linux quando tinha
HDD, mas quando troquei para SSD fiquei com pouco espaço para um dual boot
e tenho que me virar com Vagrant/Docker e VirtualBox para rodar projetos em
Linux.

Em breve vou colocar um HDD pra rodar Linux em dual boot novamente, e poder
usar Linux no dia a dia, não só para projetos.

2017-09-12 10:21 GMT-03:00 Ruben Marcus <notifications@github.com>:
